src/System.Net.Security/tests/FunctionalTests/SslStreamStreamToStreamTest.cs

```diff
@@ -28,9 +29,16 @@ public void SslStream_StreamToStream_Authentication_Success()
                 Task[] auth = new Task[2];
                 auth[0] = client.AuthenticateAsClientAsync(certificate.Subject);
                 auth[1] = server.AuthenticateAsServerAsync(certificate);
-
+                
                 bool finished = Task.WaitAll(auth, TimeSpan.FromSeconds(3));
                 Assert.True(finished, "Handshake completed in the allotted time");
+
+                string testData = "Hello SSL";
+                byte[] sendBuf = Encoding.UTF8.GetBytes(testData);
+                client.Write(sendBuf);
+                byte[] recvBuf = new byte[sendBuf.Length];
+                server.Read(recvBuf, 0, recvBuf.Length);
```

Comment:
> Is this true for a very small data

In general it's true for any amount of data.  A conforming Stream.Read implementation could return 1 byte regardless of how many more than that you ask for.